res/enkf/es_update.py

```diff
@@ -36,48 +37,45 @@ class SmootherSnapshot:
     update_step_snapshots: Dict[str, UpdateSnapshot] = field(default_factory=dict)
 
 
-def analysis_smoother_update(
+def _create_matrices(
```

Comment:
Not sure, but think it might be better to violate the `DRY` here and get rid of this function.

Reply:
Fixed it to DRY and resulted in fewer lines :sweat_smile: